app/views/hosts/_form.html.erb

```diff
@@ -139,6 +139,6 @@
 
     <%= hidden_field_tag 'bare_metal_capabilities', @host.bare_metal_capabilities, :disabled => true %>
     <%= f.hidden_field :overwrite? %>
-    <%= submit_or_cancel f, @host.overwrite?, :cancel_path => resource_prev_url_with_search_filters || (@host.new_record? ? current_hosts_path : current_host_details_path(@host)) %>
+    <%= submit_or_cancel f, @host.overwrite?, :cancel_path => resource_prev_url_with_search_filters || (@host.new_record? ? current_hosts_path : (@host.name.present? ? current_host_details_path(@host) : current_host_details_path(@host.id))) %>
```

Comment:
The cancel_path logic is getting more and more complex,
WDYT about extracting it into a helper method, so we keep the logic out of the views?

Reply:
Honestly, I'd simplify this to **never** rely on the hostname. I know we use it all over the UI, but it's a pain because the hostname may change so the URL isn't persistent.

In other words:
```suggestion
    <%= submit_or_cancel f, @host.overwrite?, :cancel_path => resource_prev_url_with_search_filters || (@host.new_record? ? current_hosts_path : current_host_details_path(@host.id)) %>
```

Or if you can, make `current_host_details_path` deal with a host that doesn't have a name, but that's a Rails helper and I don't know how easy that is.